README.adoc

```diff
@@ -174,24 +228,101 @@ Same as in the `ne4j` repo.
 Suggestions for improvement::
 
 Give a possible solution for the provided example query.
+====
 
-⛔Example:
+For example:
+
+[options="header",cols="h,2,2",]
+|===
+| Example
+| ⛔ Old example
+| ✅ New example
 
+| Notification example
+a| .Cartesian product
+====
+Query::
++
+[source, cypher, role="noplay"]
+----
+MATCH (c:Child), (p:Parent) RETURN c, p
+----
+
+Description of the returned code::
+If a part of a query contains multiple disconnected patterns,
+this will build a cartesian product between all those parts.
+This may produce a large amount of data and slow down query processing.
+While occasionally intended, it may often be possible to reformulate the query that avoids the use of this cross product,
+perhaps by adding a relationship between the different parts or by using `OPTIONAL MATCH` (identifier is: (`p`))
+
+Suggestions for improvement::
 In case a cartesian product is needed, nothing can be done to improve this query.
 In many cases, however, you might not need a combination of all children and parents, and that is when this query could be improved.
 If for example, you need the children and the children's parents, you can improve this query by rewriting it to the following:
-
++
 [source, cypher, role="noplay"]
 ----
 MATCH (c:Child)-[:ChildOf]->(p:Parent) RETURN c, p
 ----
+====
+a| .Cartesian product
+====
+Query::
++
+[source, cypher, role="noplay"]
+----
+MATCH (c:Child), (p:Parent) RETURN c, p
+----
 
+Description of the returned code::
+The disconnected patterns `$param` build a cartesian product. A cartesian product may produce a large amount of data and slow down query processing. See Status Codes documentation for suggestions.
 
-✅Rewrite:
-
+Suggestions for improvement::
 If you only need the children and the children's parents, and not all combinations between them, add `[:ChildOf]` between the `Child` and the `Parent` nodes:
 
 [source, cypher, role="noplay"]
 ----
 MATCH (c:Child)-[:ChildOf]->(p:Parent) RETURN c, p
-====
\ No newline at end of file
+----
+====
+
+| Error example
+a| .Using a list range operator `[ ]` without a start or an end value.
+====
+Query::
++
+[source, cypher, role="noplay"]
+----
+RETURN range(0,10)[..]
+----
+
+Description of the returned code::
+The statement is syntactically valid, but expresses something that the database cannot do.
```

Comment:
If we are planning on adding error examples to this docs manual, I think we should omit this for them as the description is connected to the Neo4j status code and thus already documented above the example